pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/action/Actions.scala

```diff
@@ -82,14 +82,14 @@ trait AuthActions {
     */
   def authCallbackUrl: String
 
-  val OAuth = new OAuth(settings.oAuthSettings, system, authCallbackUrl)(ec, wsClient)
+  lazy val OAuth = new OAuth(settings.oAuthSettings, system, authCallbackUrl)(ec, wsClient)
 
   /**
     * Application name used for initialising Google API clients for directory group checking
     */
-  val applicationName: String = s"pan-domain-authentication-$system"
+  lazy val applicationName: String = s"pan-domain-authentication-$system"
```

Comment:
The fields `OAuth` and `applicationName` are both made `lazy` because they both reference `system` - and to evaluate `system`, they need to evaluate `panDomainSettings`, which we want to delay until necessary, due to abstract `val`s in traits [evaluating to `null` during the construction](https://docs.scala-lang.org/tutorials/FAQ/initialization-order.html).